### DISCUSSION TITLE
Having multiple simlayer map definitions for a single key triggers the first one

### DISCUSSION THREAD

**Discussion by kaygdotorg** [evan-liu/karabiner.ts#129] [Q&A] [answered]
Hello,

Thanks for your help so far. I have been using more and more of Karabiner Elements, thanks to your project. However, I have run into this minor issue where:

```
  simlayer('spacebar', 'Navigation Layer').manipulators([
    // window navigation
    map('tab').to$('open raycast://extensions/raycast/navigation/switch-windows'),
    // common MacOS shortcuts
    map('a').to('a', 'left_command'),
    map('s').to('s', 'left_command'),
    map('f').to('f', 'left_command'),
    map('t').to('t', 'left_command'),
    // copy cut paste
    map('c').to('c', 'left_command'),
    map('v').to('v', 'left_command'),
    map('x').to('x', 'left_command'),
    // more hjkl - select words
    map('h', 'right_shift').to('left_arrow', 'left_option'),
    // hjkl vim style
    map('h').to('left_arrow'),
    map('j').to('down_arrow'),
    map('k').to('up_arrow'),
    map('l').to('right_arrow'),
  ]),
```

typing `Space + h` triggers `left_arrow + left_option` for the first time but if I press h after keeping Space held, it results in just `left_arrow`. 

If I move `map('h', 'right_shift').to('left_arrow', 'left_option'),` after `map('h').to('left_arrow'),` then `left_arrow + left_option` never get triggered. Am I missing something here?

Thank you!

**Comment by kaygdotorg** [comment]
https://github.com/evan-liu/karabiner.ts/assets/30401343/95cfd158-4e28-4802-835b-85ab60ea26bb

**Reply by kaygdotorg** [reply]
As you can see, when I press `Space + h` first → `left_arrow + left_option` are triggered (which is unexpected) and on subsequent presses of `h`, the desired behaviour is triggered: `left_arrow`.

**Comment by evan-liu** [comment, marked as answer]
```typescript
map('h', 'right_shift').to('left_arrow', 'left_option'),
```

This doesn't work with `simlayer`, which needs each key on the layer to be pressed together with the layer key. 

If you want to have mappings more than that, you can use the layer condition variable instead: 

```typescript

  rule('More on navi layer').condition(ifVar('navi-layer')).manipulators([
    // Put it above `navi-layer` so Karabiner runs it instead of 
    // `right_shift` + `h` on the simlayer
    map('h', 'right_shift').to('left_arrow', 'left_option'),
  ]),

  simlayer('spacebar', 'navi-layer').manipulators([
    // map().to(), only one-key in map(), without modifiers
  ]),
```

**Reply by kaygdotorg** [reply]
> This doesn't work with simlayer, which needs each key on the layer to be pressed together with the layer key.

It does work when I press `Space + right shift + h`. The problem is that just the first time I press `Space + h`, it triggers the incorrect action. 

> If you want to have mappings more than that, you can use the layer condition variable instead:

Your solution works the expected way though. Thank you so much for that!
